### PR TITLE
Enhance --number-of-docs feature help text and error handling

### LINKED ISSUE #289: [FEATURE] Add support for number of documents when creating workloads
**Is your feature request related to a problem? Please describe.**
`create-workload` subcommand currently generates a workload using all the documents from appointed indices. It would be nice if users could specify a number of documents from the indices they've appointed instead of using all the documents.

Currently, this is what `create-workload` supports:
```
usage: opensearch-benchmark create-workload [-h] --workload WORKLOAD --indices INDICES --target-hosts TARGET_HOSTS [--client-options CLIENT_OPTIONS]
                                            [--output-path OUTPUT_PATH] [--quiet] [--offline]

options:
  -h, --help            show this help message and exit
  --workload WORKLOAD   Name of the generated workload
  --indices INDICES     Comma-separated list of indices to include in the workload
  --target-hosts TARGET_HOSTS
                        Comma-separated list of host:port pairs which should be targeted
  --client-options CLIENT_OPTIONS
                        Comma-separated list of client options to use. (default: timeout:60)
  --output-path OUTPUT_PATH
                        Workload output directory (default: workloads/)
  --quiet               Suppress as much as output as possible (default: false).
  --offline             Assume that Benchmark has no connection to the Internet (default: false).
```

**Describe the solution you'd like**
There should be an additional parameter that takes in the number of documents of appointed indices to generate workload from.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by IanHoang**
### Description
Piggy-backing off of this PR that was merged in earlier today #291. These changes introduce clearer help text and code snippets as well as better error handling. 

### Issues Resolved
#289 

### Testing
- [x] New functionality includes testing
- Manually tested several scenarios and executed tests
- Manually tested several edge cases, such as when users pass in non-existent indices, inputs that do not conform to k:v pair structure, and more. 

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: osbenchmark/utils/opts.py [hunk 1]
```diff
@@ -127,13 +128,19 @@ def __init__(self, option_strings, dest, nargs=None, **kwargs):
...
-        setattr(namespace, self.dest, my_dict)
-
-        return my_dict
+            try:
+                k,v = kv.split(":")
+                custom_dict[k] = v
+            except ValueError:
+                raise exceptions.InvalidSyntax(
+                    "StoreKeyPairAsDict: Could not convert string to dict due to invalid syntax. If using parameter " +
+                    "--number-of-docs, ensure input follows format: " + "<index1>:<doc_count1> <index2>:<doc_count2> ... "
```
Comment (gkamat) [inline osbenchmark/utils/opts.py:139]: This is helpful, but it ties the message to the --number-of-docs option.  If this action gets re-used for some other option, admittedly unlikely for now, it might need to be made more general.

File: osbenchmark/workload_generator/workload_generator.py [hunk 2]
```diff
@@ -67,10 +75,20 @@ def extract_mappings_and_corpora(client, output_path, indices_to_extract, number
...
-        docs_to_extract = int(docs_to_extract) if docs_to_extract is not None else docs_to_extract
-        c = corpus.extract(client, output_path, i["name"], docs_to_extract)
+        custom_docs_to_extract = None
+
+        if docs_were_requested and i["name"] in indices_docs_map:
+            try:
+                custom_docs_to_extract = int(indices_docs_map.get(i["name"]))
+            except ValueError:
+                raise exceptions.InvalidSyntax(
+                    f"Value in <index>:<doc_count> pair is not a string-represented integer: [{indices_docs_map.get(i['name'])}]"
```
Comment (gkamat) [inline osbenchmark/workload_generator/workload_generator.py:87]: "not the string representation of an integer" or "the string XYZ cannot be converted to an integer"

**Review by gkamat — APPROVED** [review]

File: osbenchmark/utils/opts.py (hunk 1 above)
Comment (IanHoang) [reply osbenchmark/utils/opts.py:139]: This came to mind when creating this message. I'll simplify it and make it more generic.

File: osbenchmark/workload_generator/workload_generator.py (hunk 2 above)
Comment (IanHoang) [reply osbenchmark/workload_generator/workload_generator.py:87]: Will update this. I like this more.
